# HG changeset patch
# User Doug Stewart <dastew@sympatico.ca>
# Date 1232087239 -3600
#      Fri Jan 16 07:27:19 2009 +0100
# Branch release-3-0-x
# Node ID 66881d20101d5a8b552367faf4f5c53cf5161058
# Parent  f2af2233ce7fa10e70aba8ff9425549c7e95df44
grid.m: handle minor grid option

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-12-24  Doug Stewart  <dastew@sympatico.ca>
+
+	* plot/grid.m: Handle "minor" option.
+
 2008-03-04  Ben Abbott <bpabbott@mac.com>
 
 	* polynomial/polyfit.m: Modified tests to respect a relative tolerance.
 
 2008-02-19  Ben Abbott  <bpabbott@mac.com>
 
 	* polynomial/polyfit.m: Use QR decomposition.  Handle
 	normalization option.
diff --git a/scripts/plot/grid.m b/scripts/plot/grid.m
--- a/scripts/plot/grid.m
+++ b/scripts/plot/grid.m
@@ -37,32 +37,44 @@
 
 function grid (varargin)
 
   persistent grid_on = false;
   persistent minor_on = false;
 
   [ax, varargin, nargs] = __plt_get_axis_arg__ ("grid", varargin{:});
 
-  if (nargs > 1)
+  if (nargs > 2)
     print_usage ();
   elseif (nargs == 0)
     grid_on = ! grid_on;
   else
     x = varargin{1};
     if (ischar (x))
       if (strcmp ("off", x))
 	grid_on = false;
       elseif (strcmp ("on", x))
 	grid_on = true;
       elseif (strcmp ("minor", x))
-	minor_on = ! minor_on;
-	if (minor_on)
-	  grid_on = true;
-	endif
+        if (nargs == 2)
+          x2 = varargin{2};
+          if (strcmpi (x2, "on"))
+            minor_on = true;
+            grid_on=true;
+          elseif (strcmpi (x2, "off"))
+            minor_on = false;
+          else
+            print_usage ();
+          endif
+        else
+          minor_on = ! minor_on
+          if (minor_on)
+            grid_on = true;
+          endif
+        endif
       else
 	print_usage ();
       endif
     else
       error ("grid: argument must be a string");
     endif
   endif
 
